### ISSUE TITLE
Low PQ Things Accuracy on Custom Dataset

### ISSUE THREAD

**Issue by Sylva-Lin** [tue-mps/eomt#56]
Hello, I have a question about training the model.

I trained my own dataset using EOMT, with the COCO configuration file `eomt_small_640_2x.yaml`, where I modified the `stuff_classes` to match my own categories.

<img width="934" height="498" alt="Image" src="https://github.com/user-attachments/assets/110adc97-8b0c-4005-862a-4d8eca14b0c6" />

During training, the PQ Stuff achieves high accuracy, but the PQ Things remains consistently low, and I’m not sure what’s causing this issue.

<img width="726" height="377" alt="Image" src="https://github.com/user-attachments/assets/38c9f45a-d378-4665-8c20-20003b4ac7bc" />

My training set contains 7,262 images, and the validation set contains 1,816 images. I’m training on a single RTX 4090 GPU with a batch size of 4. I used the pretrained model `dinov3-vits16-pretrain-lvd1689m`. I have a total of 18 classes, including 10 Things and 8 Stuff.

I look forward to your reply.

**Comment by tommiekerssies** [comment]
Hi @Sylva-Lin , could you try starting from a pretrained EoMT to see if that helps (in case of DINOv3 make sure to set delta_weights to True)? Also, could you try using DINOv2 as well? Furthermore, considering your small batch size, are you using gradient accumulation to get a total effective batch size of 16?

**Comment by Sylva-Lin** [comment]
@tommiekerssies Thank you for your quick reply.

Do you mean using the COCO pretrained EoMT model when you say pretrained EoMT?

I previously tried testing COCO using the `pytorch_model.bin` in `inference.ipynb`, but it raised an error. It seems that the currently available pretrained model might not be fully compatible with the latest code, so I wasn’t able to run the test.

<img width="1259" height="738" alt="Image" src="https://github.com/user-attachments/assets/7696cd67-4779-47af-8162-fdf63089fa06" />

Regarding `delta_weights`, I mentioned a similar issue in another issue:https://github.com/tue-mps/eomt/issues/55, and one of your colleagues said it could be set to False.

As for the batch size, I’m currently using 4, which occupies about 11 GB of GPU memory. I can try increasing it to 8 at most.

**Comment by NiccoloCavagnero** [comment]
Hi @Sylva-Lin,

Please carefully read my previous answer on your other issue: `delta_weights` must be `True` if you are loading the provided EoMT checkpoints with DINOv3 backbones (as Tommie is saying) and has to be `False` otherwise.

Best,
Nick

**Comment by NiccoloCavagnero** [comment]
If you are loading original DINOv3 weights you can keep `delta_weights` to `False`.

Please, try to follow what Tommie suggested, if this doesn’t help it is likely that the issue is on your side.

I would also point out that you should adjust the annealing steps according to the number of steps of you training.

Best,
Nick
